corehq/motech/management/commands/create_repeat_records.py

```diff
@@ -66,11 +67,23 @@ def _get_repeaters(doc):
                     by_domain[doc.domain] = repeaters
                 return by_domain[doc.domain]
 
-        accessor = FormAccessorSQL.get_form if options['doc_type'] == 'form' else CaseAccessorSQL.get_case
-        for doc_id in with_progress_bar(doc_ids):
+        def _create_records(doc):
+            for repeater in _get_repeaters(doc):
+                repeater.register(doc)
+
+        bulk_accessor = FormAccessorSQL.get_forms if options['doc_type'] == 'form' else CaseAccessorSQL.get_cases
+        single_accessor = FormAccessorSQL.get_form if options['doc_type'] == 'form' else CaseAccessorSQL.get_case
+        for doc_ids in chunked(with_progress_bar(doc_ids), 100):
+            doc_ids = list(doc_ids)
             try:
-                form_or_case = accessor(doc_id)
-                for repeater in _get_repeaters(form_or_case):
-                    repeater.register(form_or_case)
+                docs = bulk_accessor(doc_ids)
+                for doc in docs:
+                    _create_records(doc)
```

Comment:
It seems possible that `_create_records(doc)` could raise an exception, which would then trigger the fall-back to individual fetches. Some records may have already been created, which means they will be created a second time. Would that be a problem?

Reply:
I don't think so since repeaters are meant to be idempotent but I could keep track of the records that have been sent and skip over them.